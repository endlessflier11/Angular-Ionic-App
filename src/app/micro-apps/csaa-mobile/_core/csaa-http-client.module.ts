import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
