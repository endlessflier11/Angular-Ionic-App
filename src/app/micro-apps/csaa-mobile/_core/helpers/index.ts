export * from './router.helpers';
export * from './misc.helpers';
export * from './async.helper';
export * from './error.helper';
export * from './string.helper';
