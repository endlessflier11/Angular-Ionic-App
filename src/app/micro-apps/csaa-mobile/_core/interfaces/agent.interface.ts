export interface Agent {
  name: string;
  phone: string;
  email: string;
  isService: boolean;
}
