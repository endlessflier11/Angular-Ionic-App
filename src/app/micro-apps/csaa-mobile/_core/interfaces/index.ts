export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
