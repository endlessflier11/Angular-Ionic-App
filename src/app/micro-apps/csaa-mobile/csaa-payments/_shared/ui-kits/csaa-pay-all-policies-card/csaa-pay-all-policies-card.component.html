<csaa-skeleton-card *ngIf="loading"></csaa-skeleton-card>
<ion-card
  *ngIf="!loading && payments && payments.length > 0 && !paymentSucceeded"
  [class.overdue]="hasOverDuePayment"
  class="csaa-card"
>
  <ion-card-header>
    <ion-card-title>Pay All Policies</ion-card-title>
  </ion-card-header>
  <ion-card-content class="csaa-item-list">
    <ion-list>
      <ion-item detail class="csaa-payment-amount-list-item" (click)="selectAmount()">
        <ion-label>Amount</ion-label>
        <span slot="end">{{ amount | currency: 'USD':'symbol':'1.2-2' }}</span>
      </ion-item>

      <ion-item detail lines="none" class="csaa-payment-method-list-item" (click)="selectPayment()">
        <ion-avatar slot="start" *ngIf="paymentMethod">
          <span>
            <!-- Span first because ion-avatar::slotted(img) style forces image to render unexpectedly -->
            <img
              *ngIf="paymentMethod?.card?.type"
              [src]="'/assets/csaa-mobile/vectors/' + paymentMethod.card.type + '.svg'"
            />
            <img *ngIf="paymentMethod?.account" [src]="'/assets/csaa-mobile/vectors/bank.svg'" />
          </span>
        </ion-avatar>

        <ion-label
          [ngClass]="{ 'card-expired': paymentMethod?.card && paymentMethod?.card.expired }"
          class="ion-text-nowrap"
        >
          <span>{{ paymentMethod ? paymentMethod.shortName : 'Pay With' }}</span>
        </ion-label>
        <span slot="end" *ngIf="paymentMethod?.card && paymentMethod?.card?.isDebitCard !== true">
          {{ 'Credit' }}
        </span>
        <span slot="end" *ngIf="paymentMethod?.card?.isDebitCard">{{ 'Debit' }}</span>
        <span slot="end" *ngIf="paymentMethod?.account">{{
          paymentMethod?.account?.type | titlecase
        }}</span>
        <span slot="end" *ngIf="!paymentMethod">Payment Method</span>
      </ion-item>
    </ion-list>

    <ion-button
      [disabled]="isButtonDisabled()"
      expand="full"
      (click)="makePayment()"
      class="ion-no-margin"
    >
      Make a Payment
    </ion-button>
  </ion-card-content>
</ion-card>
