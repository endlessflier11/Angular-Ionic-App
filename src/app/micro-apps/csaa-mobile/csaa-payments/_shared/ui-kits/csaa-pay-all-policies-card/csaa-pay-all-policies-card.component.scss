@import '../../../../_scss/variables';
.csaa-card {
  ion-card-content {
    ion-list {
      margin-bottom: 20px;
    }

    ion-item.csaa-payment-amount-list-item {
      font-size: 16px !important;
      > span {
        margin-inline-end: 5px;
        color: $csaa-lighter-color;
      }
    }

    ion-item.csaa-payment-method-list-item {
      font-size: 16px !important;

      ion-avatar {
        margin-inline-end: 5px;
        > span {
          width: 30px;
          display: flex;
          align-items: center;
          height: 100%;
        }

        img {
          max-width: 100%;
        }
      }

      ion-label {
        margin-inline-end: 5px;
        > span {
          display: block;
          width: 100%;
          text-overflow: ellipsis;
          overflow: hidden;
        }
      }

      > span {
        color: $csaa-lighter-color;
        margin-inline-start: 10px;
        margin-inline-end: 5px;
      }
    }
  }

  &.overdue {
    ion-card-title {
      color: $csaa-emergency-color;
    }
    ion-card-content {
      ion-item.csaa-payment-amount-list-item {
        > ion-label,
        > span {
          color: $csaa-emergency-color;
        }
      }
    }
  }
}
