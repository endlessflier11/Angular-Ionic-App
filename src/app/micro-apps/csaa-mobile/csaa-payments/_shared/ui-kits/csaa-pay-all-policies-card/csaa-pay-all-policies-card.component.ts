import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
import {
  Category,
  EventName,
  EventType,
  PaymentAccount,
  PaymentType,
  UpcomingPayment,
  UpcomingPaymentModel,
} from '../../../../_core/interfaces';
import { AnalyticsService, parseToCent } from '../../../../_core/services';

@Component({
  selector: 'csaa-pay-all-policies-card',
  templateUrl: 'csaa-pay-all-policies-card.component.html',
  styleUrls: ['csaa-pay-all-policies-card.component.scss'],
})
export class CsaaPayAllPoliciesCardComponent implements OnChanges {
  @Input() paymentSucceeded = false;
  @Input() timezone: string = undefined;
  @Input() payments: UpcomingPayment[];
  @Input() paymentMethod: PaymentAccount;
  @Input() disabled = false;
  @Input() loading = false;

  @Output() selectAmountClick = new EventEmitter<UpcomingPayment>();
  @Output() selectPaymentClick = new EventEmitter<UpcomingPayment>();
  @Output() makePaymentClick = new EventEmitter<UpcomingPayment>();

  public combinedPayment: UpcomingPayment;
  public hasOverDuePayment: boolean;

  constructor(private readonly analyticsService: AnalyticsService) {}

  ngOnChanges(changes: SimpleChanges) {
    if (
      changes &&
      changes.payments &&
      changes.payments.currentValue &&
      changes.payments.currentValue.length > 0
    ) {
      this.hasOverDuePayment = this.checkForOverDuePayment();
      this.combinedPayment = this.createPayment();
    }
  }

  createPayment() {
    const combinedPayment = Object.assign({}, this.payments[0]);
    combinedPayment.allPolicies = true;
    combinedPayment.policyNumber = 'ALL';
    combinedPayment.minimumAmount = 0;
    combinedPayment.remainingPremium = 0;

    this.payments.forEach((currentPayment) => {
      if (currentPayment.remainingPremium > 0) {
        combinedPayment.remainingPremium += parseToCent(currentPayment.remainingPremium);
      }

      if (currentPayment.minimumAmount > 0) {
        combinedPayment.minimumAmount += parseToCent(currentPayment.minimumAmount);
      }
    });

    // Revert from cent since we completed calculation
    combinedPayment.minimumAmount = combinedPayment.minimumAmount / 100;
    combinedPayment.remainingPremium = combinedPayment.remainingPremium / 100;

    return new UpcomingPaymentModel(combinedPayment);
  }

  get remaining() {
    return this.combinedPayment.remainingPremium;
  }

  get amount() {
    return this.combinedPayment.amount;
  }

  checkForOverDuePayment(): boolean {
    return this.payments
      ? !!this.payments.find((payment: UpcomingPayment) => payment.isPaymentDue)
      : false;
  }

  setPaymentAmountType(type: PaymentType): void {
    this.combinedPayment.type = type;
  }

  selectAmount(): void {
    this.selectAmountClick.emit(this.combinedPayment);
  }

  selectPayment(): void {
    this.selectPaymentClick.emit(this.combinedPayment);
  }

  makePayment() {
    this.analyticsService.trackEvent(EventName.PAY_ALL_SELECTED, Category.payments, {
      event_type: EventType.OPTION_SELECTED,
      selection: 'Pay All',
      amount_due: this.amount,
      ...AnalyticsService.mapPaymentPolicy(...this.payments),
    });
    this.makePaymentClick.emit(this.combinedPayment);
  }

  isButtonDisabled() {
    return (
      this.disabled ||
      !this.amount ||
      !this.paymentMethod ||
      (this.paymentMethod.card && this.paymentMethod.card.expired) ||
      this.amount < 0
    );
  }
}
