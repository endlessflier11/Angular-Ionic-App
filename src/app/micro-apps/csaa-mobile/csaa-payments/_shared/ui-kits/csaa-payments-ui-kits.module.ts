import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaPaymentHistoryCardComponent } from './csaa-payment-history-card/csaa-payment-history-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { CsaaPaymentHistorySelectCardComponent } from './csaa-payment-history-select-card/csaa-payment-history-select-card.component';
import { CsaaPaymentSucceededCardComponent } from './csaa-payment-succeeded-card/csaa-payment-succeeded-card.component';
import { CsaaMakePaymentCardComponent } from './csaa-make-payment-card/csaa-make-payment-card.component';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card/csaa-pay-all-policies-card.component';
import { PaymentTermsConditionsComponent } from './payment-terms-conditions/payment-terms-conditions.component';
import { AutopayTermsConditionsComponent } from './autopay-terms-conditions/autopay-terms-conditions.component';
import { CsaaBillsCardComponent } from './csaa-bills-card/csaa-bills-card.component';
import { AutopayDiscountCardComponent } from './autopay-discount-card/autopay-discount-card.component';

const SHARED_COMPONENTS = [
  CsaaPaymentHistoryCardComponent,
  CsaaPaymentHistorySelectCardComponent,
  CsaaMakePaymentCardComponent,
  CsaaPaymentSucceededCardComponent,
  CsaaPayAllPoliciesCardComponent,
  PaymentTermsConditionsComponent,
  AutopayTermsConditionsComponent,
  CsaaBillsCardComponent,
  AutopayDiscountCardComponent,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule, IonicModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaPaymentsUiKitsModule {}
