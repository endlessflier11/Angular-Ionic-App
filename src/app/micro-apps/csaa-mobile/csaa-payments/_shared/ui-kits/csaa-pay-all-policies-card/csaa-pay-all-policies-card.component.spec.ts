import { CommonModule } from '@angular/common';
import { NO_ERRORS_SCHEMA, SimpleChanges } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule } from '@ionic/angular';
import { By } from '../../../../../../../testing';
import { PageTestingModule } from '../../../../../../../testing/misc/page-testing.module';
import { Category, EventName, PolicyType } from '../../../../_core/interfaces';
import { PaymentType } from '../../../../_core/interfaces/payment.interface';
import { AnalyticsService } from '../../../../_core/services/analytics.service';
import DateHelper from '../../../../_core/shared/date.helper';
import { UiKitsModule } from '../../../../_core/ui-kits/ui-kits.module';
import { CsaaPayAllPoliciesCardComponent } from './csaa-pay-all-policies-card.component';

describe('CsaaPayAllPoliciesCardComponent', () => {
  let component: CsaaPayAllPoliciesCardComponent;
  let fixture: ComponentFixture<CsaaPayAllPoliciesCardComponent>;
  let analyticsService;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [CsaaPayAllPoliciesCardComponent],
      imports: [
        CommonModule,
        IonicModule,
        UiKitsModule,
        PageTestingModule.withConfig({ providesStorage: true, providesAnalytics: true }),
      ],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();

    analyticsService = TestBed.inject(AnalyticsService);
    fixture = TestBed.createComponent(CsaaPayAllPoliciesCardComponent);
    component = fixture.componentInstance;
    component.timezone = '+0000';
    fixture.detectChanges();
  });
  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshots when payment is autoPay and there are overdue payments', () => {
    const payments = [
      {
        vehicles: ['Lexus NX', 'BMW X1'],
        subtitle: 'test',
        autoPay: true,
        minimumAmount: 20,
        remainingPremium: 10000,
        otherAmount: 1000,
        type: PaymentType.other,
        nextDueDate: DateHelper.toDate('2040-08-19'),
        nextAmount: 30,
        allPolicies: false,
        policyNumber: 'AZSS294857564',
        policyType: PolicyType.Auto,
        dueDate: DateHelper.toDate('2018-08-10'),
        isPastDue: true,
        isPaymentDue: true,
        amount: 260.75,
        policyRiskState: 'test',
        autopayEnrollment: null,
      },
    ];

    component.payments = payments;
    component.ngOnChanges({ payments: { currentValue: payments } } as any as SimpleChanges);
    fixture.detectChanges();
    component.paymentSucceeded = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('it should display selectAmountClick when clicking on card details', () => {
    const payments = [
      {
        vehicles: ['Lexus NX', 'BMW X1'],
        subtitle: 'test',
        autoPay: true,
        minimumAmount: 20,
        remainingPremium: 10000,
        otherAmount: 1000,
        type: PaymentType.other,
        nextDueDate: DateHelper.toDate('2040-08-19'),
        nextAmount: 30,
        allPolicies: false,
        policyNumber: 'AZSS294857564',
        policyType: PolicyType.Auto,
        dueDate: DateHelper.toDate('2021-08-10'),
        isPastDue: false,
        isPaymentDue: false,
        amount: 260.75,
        policyRiskState: 'test',
        autopayEnrollment: null,
      },
    ];
    component.payments = payments;
    component.ngOnChanges({ payments: { currentValue: payments } } as any as SimpleChanges);

    fixture.detectChanges();
    const button = fixture.debugElement.query(By.css('.csaa-payment-amount-list-item'));
    button.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('it should display selectPaymentClick when clicking on card details', () => {
    const payments = [
      {
        vehicles: ['Lexus NX', 'BMW X1'],
        subtitle: 'test',
        autoPay: true,
        minimumAmount: 20,
        remainingPremium: 10000,
        otherAmount: 1000,
        type: PaymentType.other,
        nextDueDate: DateHelper.toDate('2040-08-19'),
        nextAmount: 30,
        allPolicies: false,
        policyNumber: 'AZSS294857564',
        policyType: PolicyType.Auto,
        dueDate: DateHelper.toDate('2021-08-10'),
        isPastDue: false,
        isPaymentDue: false,
        amount: 260.75,
        policyRiskState: 'test',
        autopayEnrollment: null,
      },
    ];
    component.payments = payments;
    component.ngOnChanges({ payments: { currentValue: payments } } as any as SimpleChanges);
    fixture.detectChanges();
    const button = fixture.debugElement.query(By.css('.csaa-payment-method-list-item'));
    button.triggerEventHandler('click', null);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should track the make payment button', () => {
    const payments = [
      {
        vehicles: ['Lexus NX', 'BMW X1'],
        subtitle: 'test',
        autoPay: true,
        minimumAmount: 20,
        remainingPremium: 10000,
        otherAmount: 1000,
        type: PaymentType.other,
        nextDueDate: DateHelper.toDate('2040-08-19'),
        nextAmount: 30,
        allPolicies: false,
        policyNumber: 'AZSS294857564',
        policyType: PolicyType.Auto,
        dueDate: DateHelper.toDate('2021-08-10'),
        isPastDue: false,
        isPaymentDue: false,
        amount: 260.75,
        policyRiskState: 'test',
        autopayEnrollment: null,
      },
    ];
    component.payments = payments;
    component.ngOnChanges({ payments: { currentValue: payments } } as any as SimpleChanges);
    fixture.detectChanges();
    component.makePayment();
    expect(analyticsService.trackEvent).toHaveBeenCalledWith(
      EventName.PAY_ALL_SELECTED,
      Category.payments,
      {
        event_type: 'Option Selected',
        selection: 'Pay All',
        amount_due: component.amount,
        policies: [
          {
            policy_number: 'AZSS294857564',
            policy_state: 'test',
            policy_type: 'Auto',
          },
        ],
      }
    );
  });
  it('it should match snapshots right after submitting a successful payment', () => {
    const payments = [
      {
        vehicles: ['Lexus NX', 'BMW X1'],
        subtitle: 'test',
        autoPay: true,
        minimumAmount: 20,
        remainingPremium: 10000,
        otherAmount: 1000,
        type: PaymentType.other,
        nextDueDate: DateHelper.toDate('2040-08-19'),
        nextAmount: 30,
        allPolicies: false,
        policyNumber: 'AZSS294857564',
        policyType: PolicyType.Auto,
        dueDate: DateHelper.toDate('2021-08-10'),
        isPastDue: false,
        isPaymentDue: false,
        amount: 260.75,
        policyRiskState: 'test',
        autopayEnrollment: null,
      },
    ];
    component.payments = payments;
    component.ngOnChanges({ payments: { currentValue: payments } } as any as SimpleChanges);
    fixture.detectChanges();
    component.paymentSucceeded = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
