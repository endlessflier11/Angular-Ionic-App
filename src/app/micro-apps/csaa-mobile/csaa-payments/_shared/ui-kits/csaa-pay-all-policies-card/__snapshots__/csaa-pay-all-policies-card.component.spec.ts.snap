// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaPayAllPoliciesCardComponent it should display selectAmountClick when clicking on card details 1`] = `
<csaa-pay-all-policies-card
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  combinedPayment={[Function UpcomingPaymentModel]}
  disabled="false"
  hasOverDuePayment="false"
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  paymentSucceeded="false"
  payments={[Function Array]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Pay All Policies
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item"
          detail=""
        >
          <ion-label>
            Amount
          </ion-label>
          <span
            slot="end"
          >
            $20.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-method-list-item"
          detail=""
          lines="none"
        >
          <ion-label
            class="ion-text-nowrap"
          >
            <span>
              Pay With
            </span>
          </ion-label>
          <span
            slot="end"
          >
            Payment Method
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pay-all-policies-card>
`;

exports[`CsaaPayAllPoliciesCardComponent it should display selectPaymentClick when clicking on card details 1`] = `
<csaa-pay-all-policies-card
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  combinedPayment={[Function UpcomingPaymentModel]}
  disabled="false"
  hasOverDuePayment="false"
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  paymentSucceeded="false"
  payments={[Function Array]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card"
  >
    <ion-card-header>
      <ion-card-title>
        Pay All Policies
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item"
          detail=""
        >
          <ion-label>
            Amount
          </ion-label>
          <span
            slot="end"
          >
            $20.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-method-list-item"
          detail=""
          lines="none"
        >
          <ion-label
            class="ion-text-nowrap"
          >
            <span>
              Pay With
            </span>
          </ion-label>
          <span
            slot="end"
          >
            Payment Method
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pay-all-policies-card>
`;

exports[`CsaaPayAllPoliciesCardComponent it should match snapshots right after submitting a successful payment 1`] = `
<csaa-pay-all-policies-card
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  combinedPayment={[Function UpcomingPaymentModel]}
  disabled="false"
  hasOverDuePayment="false"
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  paymentSucceeded={[Function Boolean]}
  payments={[Function Array]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
</csaa-pay-all-policies-card>
`;

exports[`CsaaPayAllPoliciesCardComponent should match snapshots when payment is autoPay and there are overdue payments 1`] = `
<csaa-pay-all-policies-card
  __ngContext__={[Function LRootView]}
  analyticsService={[Function AnalyticsMockService]}
  combinedPayment={[Function UpcomingPaymentModel]}
  disabled="false"
  hasOverDuePayment={[Function Boolean]}
  loading="false"
  makePaymentClick={[Function EventEmitter_]}
  paymentSucceeded="false"
  payments={[Function Array]}
  selectAmountClick={[Function EventEmitter_]}
  selectPaymentClick={[Function EventEmitter_]}
  timezone={[Function String]}
>
  <ion-card
    class="csaa-card overdue"
  >
    <ion-card-header>
      <ion-card-title>
        Pay All Policies
      </ion-card-title>
    </ion-card-header>
    <ion-card-content
      class="csaa-item-list"
    >
      <ion-list>
        <ion-item
          class="csaa-payment-amount-list-item"
          detail=""
        >
          <ion-label>
            Amount
          </ion-label>
          <span
            slot="end"
          >
            $20.00
          </span>
        </ion-item>
        <ion-item
          class="csaa-payment-method-list-item"
          detail=""
          lines="none"
        >
          <ion-label
            class="ion-text-nowrap"
          >
            <span>
              Pay With
            </span>
          </ion-label>
          <span
            slot="end"
          >
            Payment Method
          </span>
        </ion-item>
      </ion-list>
      <ion-button
        class="ion-no-margin"
        expand="full"
      >
         Make a Payment 
      </ion-button>
    </ion-card-content>
  </ion-card>
</csaa-pay-all-policies-card>
`;
